### ISSUE TITLE
About FOCALLOSS : avg_factor

### ISSUE THREAD

**Issue by PanffeeReal** [open-mmlab/mmdetection#8337]
In the function of loss_single() of xxx_head.py ,there is avg_factor and it is passed throught  'focal_loss' to function 'weight_reduce_loss',while I found that the programm still goes into 'if avg_factor is None:' branch .
so why does it happen? please help me :
###codes are :
1. loss_cls = self.loss_cls(
            cls_score, labels, label_weights, avg_factor=num_total_samples) in anchor_head.py
2. loss = weight_reduce_loss(loss, weight, reduction, avg_factor) in focal_loss.py
3.    # if avg_factor is not specified, just reduce the loss
    if avg_factor is None:
        loss = reduce_loss(loss, reduction)  in utils.py

**Comment by PanffeeReal** [comment]
as for fsaf_head.py ,the above question happened;while for retinanet,eveything goes normally.
But fsaf_head is inherited from retinanet and there is no particular switch to control avg_factor,which is quite confusing>>><<<

**Comment by ZwwWayne** [comment]
We simply follows the official implementation to use the avg_factor.
